site/snippets/magic.ts

```diff
@@ -9,14 +9,18 @@ const MAGIC_API_KEY = "pk_test_...";
 export const magic = new Magic(MAGIC_API_KEY);
 
 // a viem wallet client that wraps magic for utility methods
-// NOTE: this isn't necessary since you can just use the `magic.rpcProvider`
-// directly, but this makes things much easier
+// NOTE: because this is async, you will need to put this in a useEffect hook if using React
 export const magicClient = createWalletClient({
-  transport: custom(magic.rpcProvider),
```

Comment:
this one doesn't work? did not look into magic code, but why is getting rpc provider async here?
hmm, and just using rpcProvider field directly seems to work fine on examples I tried.

Reply:
yea I'm not sure why this didn't work for a customer... maybe it was because you have to authenticate with a user first?

lemme take another pass at this and make it even more clear that authentication is required first

Reply:
in the unauthed case `.rpcProvider` was returning undefined

Reply:
hmm I didn't need to do this if I mapped the magicClient to `any` for the new WalletClientSigner below:

https://github.com/alchemyplatform/aa-simple-dapp/blob/a71b0b78/src/hooks/useMagicSigner.ts#L17